src/Drivers/Translation.php

```diff
@@ -51,6 +52,63 @@ public function saveMissingTranslations($language = false)
         }
     }
 
+    /**
+     * Save all of the translations in the app without translation for a given language then
+     * Translate all the tokens into it's respective language using google translate
+     *
+     * @param  string  $language
+     * @return void
+     */
+    public function autoTranslate($language = false)
+    {
+        $languages = $language ? [$language => $language] : $this->allLanguages();
+
+        foreach ($languages as $language => $name) {
+            $this->saveMissingTranslations($language);
+            $this->translateLanguage($language);
+        }
+    }
+
+    /**
+     * 
+     * Translate text using Google Translate
+     * 
+     * @param $language
+     * @param $token
+     * @return string|null
+     * @throws \ErrorException
+     */
+    public function getGoogleTranslate($language,$token){
+        $tr = new GoogleTranslate($language);
+        return $tr->translate($token);
+    }
+
+    /**
+     * 
+     * Loop through all the keys and get translated text from google 
+     * 
+     * @param $language
+     */
+    public function translateLanguage($language){
+        $translations = $this->allTranslationsFor($language);
+        foreach ($translations as $type => $groups) {
+            foreach ($groups as $group => $translations) {
+                foreach ($translations as $key => $value) {
+                    if (in_array($value,["",null])){
+                        $new_value=$this->getGoogleTranslate($language,$key);
```

Comment:
W.r.t. the issue I identified earlier,

>    Instead of translating the value in en.json, it translates the key from the source code. This causes issues if the key is not equal to the translatable text!

You could probably fix this by changing this line  to            
```php
$new_value=$this->getGoogleTranslate($language,__($key));
```

Reply:
I've created a pull request for your auto-translate branch to fix this, please check it out! 

https://github.com/timoye/laravel-translation/pull/1